### ISSUE TITLE
CJK at end of line issues

### ISSUE THREAD

**Issue by msftrncs** [PowerShell/PSReadLine#1218]
Environment data
----------------
PSReadLine 2.0.0-beta6, Windows 10, Windows PowerShell or PowerShell Core

Steps to reproduce
--------------------------------------

1. Produce a command line who's last character is a CJK doublewidth character, and the command line is just one character from wrapping.  
2. Then edit the command line to include one more character before the CJK, and the CJK will roll to the next line, but the newly inserted character is not added.  
3. Furthermore, when the extra character is removed, and the CJK character rolls back on to the first line, its not completely removed from where it was shown on the second line.

I think this might all have to do with some blanking assumptions, and the wrong spots are being blanked at the wrong time.

Demonstrating 1 and 3: (Cursor position does not correlate to the steps)
![image](https://user-images.githubusercontent.com/26179051/69911144-96aeea80-13dc-11ea-830f-6227efbcd794.png)

After step 2, there should be only 1 blank at the end of the first line (I added another 'f'):
![image](https://user-images.githubusercontent.com/26179051/69911147-b9d99a00-13dc-11ea-9b2a-2a4d6e255431.png)

Note, I have no idea what CJK character I copied and pasted when testing another issue.

**Comment by msftrncs** [comment]
It appears that recalling such a line works fine, so it only seems to occur while editing.

**Comment by msftrncs** [comment]
I think I can determine that this is a `[Console]` bug that appears in both ConsoleHost and Windows Terminal.

Example the following script, but in order for it to work, set a console up as 80 characters wide, and with a font that will show the CJK character.

```PowerShell
function test {
    [console]::Clear()
    [console]::OutputEncoding = $outputencoding
    [console]::Write(('0123456789'*7) +'01234567取')
    [console]::SetCursorPosition(0,0)
    [console]::Write(('0123456789'*7) +'012345678取')
}
```

This should produce the result of 
```PowerShell
[console]::Write(('0123456789'*7) +'012345678取')
```

but because the particular `[Console]::Write()` call overwrites an existing line, it incorrectly writes the final characters of the line.

ConsoleHost:
![image](https://user-images.githubusercontent.com/26179051/70882280-c0e3d900-1f94-11ea-95ec-3a8506d23486.png)

Windows Terminal:
![image](https://user-images.githubusercontent.com/26179051/70882515-a827f300-1f95-11ea-9971-bef09879eef1.png)

Where does this issue properly belong?  .Net Core or Windows Terminal?

**Comment by msftrncs** [comment]
I can add that this can be demonstrated in VS Code's terminal as well, so I lean towards it being a .Net Core `[Console]` issue.

Note: In VS Code, the `$OutputEncoding` had to be manually set to `[text.encoding]::utf8`, probably because I was running it in Windows PowerShell.
